README.md
# motorcortex-animetitles

## [Demo](https://kissmybutton.github.io/motorcortex-animetitles/demo/)

## Installation

```bash
$ npm install --save @kissmybutton/motorcortex-animetitles
# OR
$ yarn add @kissmybutton/motorcortex-animetitles
```

## Loading

```javascript
import {loadPlugin} from "@kissmybutton/motorcortex";
import TitlesPlugin from "@kissmybutton/motorcortex-animetitles";
const Titles = loadPlugin(TitlesPlugin);
```

# Create incident

## RotatedLine

```javascript
const nameOfIncident = new Titles.RotatedLine(
  {
    width: 400,
    size: "XL",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle1: "test",
    subTitle2: "test2",
    subTitle3: "test3",
    title: "dokimi",
    stopOnLast: false,
    delayEnd : 3000
  },
  {
    selector: ".rotatedline"
  }
);
```

### RotateLine Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width      | total width of clip container | all positive numbers |
| size |  scale of clip components   |  "S","M","L","XL"  |
| lineColor |  the color of line |  hex values or RGB  |
| textColor |  the color of text |  hex values or RGB  |
| subTitle1 , subTitle2 , subTitle3 |  right texts |  normal text  |
| title |  left text |  normal text  |
| stopOnLast |  if you like to stop animation on last text line |  true,false |
| delayEnd |  delay before last animation |  millisecond  |



## RolingText

```javascript
const nameOfIncident = new Titles.RolingText(
  {
    width: 250,
    size: "L",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle1: "test",
    subTitle2: "test2",
    subTitle3: "test3",
    stopOnLast: false,
    delayEnd : 3000
  },
  {
    selector: ".rolingtext"
  }
);
```

### Rolingtext Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width      | total width of clip container | all positive numbers |
| size |  scale of clip components   |  "S","M","L","XL","XXL","XXXL"    |
| lineColor |  the color of line |  hex values or RGB  |
| textColor |  the color of text |  hex values or RGB  |
| subTitle1 , subTitle2 , subTitle3 |  roling text lines |  normal text  |
| stopOnLast |  if you like to stop animation on last text line |  true,false |
| delayEnd |  delay before last animation |  millisecond  |



## RotatadLineReveal

```javascript
const nameOfIncident = new Titles.RotatadLineReveal(
  {
    duration: 5000,
    width: 800,
    size: "M",
    lineColor: "#ff0000",
    textColor: "#000",
    title: "Animetitle",
    subtitle: "MC Plugin",
    leftEnd: 100,
    stopOnLast: false,
    delayEnd : 3000
  },
  {
    selector: ".rotatedlinereveal"
  }
);
```

### RotatadLineReveal Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width      | total width of clip container | all positive numbers |
| size |  scale of clip components   |  "S","M","L"    |
| lineColor |  the color of line |  hex values or RGB  |
| textColor |  the color of text |  hex values or RGB  |
| title | title text |  normal text  |
| subtitle | subtitle text |  normal text  |
| leftEnd |  left distance of componet from the left side of container |  numbers |
| stopOnLast |  if you like to stop animation on last text line |  true,false |
| delayEnd |  delay before last animation |  millisecond  |



## SvgBorder

```javascript
const nameOfIncident = new Titles.SvgBorder(
  {
   duration: 5000,
    size: "L",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle: "Develop by",
    slogan: "-KMB-",
    title: "MotorCortex Team",
    stopOnLast: false,
    delayEnd : 3000
  },
  {
    selector: ".svgborder"
  }
);
```

### SvgBorder Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width      | total width of clip container | all positive numbers |
| size |  scale of clip components   |  "S","M","L" |
| lineColor |  the color of line |  hex values or RGB  |
| textColor |  the color of text |  hex values or RGB  |
| subTitle | top left text |  normal text  |
| slogan  | under border text |  normal text  |
| title  | title text |  normal text  |
| stopOnLast |  if you like to stop animation on last text line |  true,false |
| delayEnd |  delay before last animation |  millisecond  |



## Circle

```javascript
const nameOfIncident = new Titles.Circle(
  {
    fontSize: 40,
    circleColor: "#ff0000",
    textColor: "#000",
    title: "IOAN",
    subTitle: "BRAPAPAPA",
    stopOnLast: false,
    delayEnd : 3000,
    width:500
  },
  {
    selector: ".circle"
  }
);

```
## In order to use tin incident you should include the img.svg that is in demo folder in your clip.js directory
### Circle Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width      | total width of clip container and width/height of the circle | all positive numbers |
| circleColor |  the color of circle |  hex values or RGB  |
| textColor |  the color of text |  hex values or RGB  |
| subTitle | top left text |  normal text  |
| title  | svg cobe with 1 path tag only|  svg text  |
| stopOnLast |  if you like to stop animation on last text line |  true,false |
| delayEnd |  delay before last animation |  millisecond  |


## SvgDrow

```javascript
const nameOfIncident = new Titles.SvgDrow(
  {
 duration: 1600,
    width: 900,
    size: "M",
    lineColor: "#ff0000",
    strokeDashOffset: 10000,
    StrokeDashArray: 10000,
    erase: true,
    svg: `svg code`,
    stopOnLast: false,
    delayEnd : 3000
  },
  {
    selector: ".svgdrow"
  }
);
```

### SvgDrow Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width      | total width of clip container | all positive numbers |
| size |  scale of clip components   |  "S","M","L" |
| lineColor |  the color of line |  hex values or RGB  |
| textColor |  the color of text |  hex values or RGB  |
| strokeDashOffset: | The strokeDashOffset attribute is a presentation attribute defining an offset on the rendering of the associated dash array. |  number  |
| StrokeDashArray  | The StrokeDashArray attribute is a presentation attribute defining the pattern of dashes and gaps used to paint the outline of the shape; |  number  |
| erase  | erase svg after drow it |  true, false  |
| svg |  The width of the title container  |  numbers |
| stopOnLast |  if you like to stop animation on last text line |  true,false |
| delayEnd |  delay before last animation |  millisecond  |


## LogoBox

```javascript
const nameOfIncident = new Titles.LogoBox(
  {
    size: 2,
    textColor: "#fff",
    bgColor: "#000",
    subTitle1: "testdawdawddawd",
    subTitle2: "test2",
    subTitle3: "test3",
    width: 200,
    stopOnLast: false,
    logoUrl: "https://kissmybutton.github.io/motorcortex-animetitles/demo/logo.png"
  },
  {
    selector: ".logobox"
  }
);
```
## In order to use tin incident you should include the logo.png image in your clip.js directory

### LogoBox Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| size     | change the size of element  | all positive numbers |
| bgColor   |  the color of bg |  hex values or RGB  |
| textColor |  the color of text |  hex values or RGB  |
| subTitle1 , subTitle2 , subTitle3 |  right texts |  normal text  |
| width      | total width of text container | all positive numbers |
| stopOnLast |  if you like to stop animation on last text line |  true,false |
| logoUrl      | url of an image | string |


# Add incident to your clip

```javascript
clipName.addIncident(nameOfIncident, 0);

```





CHANGELOG.md
# Changelog

### [1.0.3](https://www.github.com/kissmybutton/motorcortex-animetitles/compare/v1.0.2...v1.0.3) (2021-07-01)


### Bug Fixes

* remove timing attribute from incidents ([2a9cad8](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/2a9cad84b02d4c3b2cdb5d49941eeae9cd69a5fe))

### 1.0.2 (2021-04-16)


### Features

* add delay ([8fd8f16](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/8fd8f1678cf5356ba134e69c42b92ef0e5195e56))
* new mc and player version ([cf826ee](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/cf826ee5bd93103b8648e73383813f03f94a54e0))
* update dependencies ([2f8b3c7](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/2f8b3c786a3c020c5328cf8b4f7098967f6cb6d8))


### Bug Fixes

* demo sound fix ([7fc114e](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/7fc114ea80757bcda040b9123be69dc1425116b7))
* fix audio size ([7098d40](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/7098d40259d6bb186db8eb1d80ad17b41b7e369b))
* fix circle ([b3ce61e](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/b3ce61ed9012be1703560c0065cee7466228df0c))
* fix riling text ([605ec55](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/605ec55572b743be179cdcac1f0d2c75237fc853))
* implement rollup and minor fixes ([6658270](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/66582705fd51f616ad49d410fecbfc40727be5cf))
* new version ([fd8fa1f](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/fd8fa1f8219adb5b5834962cb9cdec016295986c))


demo/index.js
import MotorCortex from "@kissmybutton/motorcortex";
import Player from "@kissmybutton/motorcortex-player";
import AnimeDefinition from "@kissmybutton/motorcortex-anime";
import TitlesPlugin from "../dist/motorcortex-animetitles.umd";
import imgsvg from "./assets/imgsvg.js";

const Anime = MotorCortex.loadPlugin(AnimeDefinition);
const Titles = MotorCortex.loadPlugin(TitlesPlugin);

const clip = new MotorCortex.HTMLClip({
  css: `
    body{ 
      background-color : white; 
    }

    .container{
      position: relative; 
      background:linear-gradient(141deg, #ccc 25%, #eee 40%, #ddd 55%);
      height:100%;
      display: flex;
      justify-content: space-around;
      align-items: center;
      flex-direction: column;
    }

    .row{
      display: flex;
      justify-content: space-around;
      align-items: center;
    }

    .cel{
      display: flex;
      justify-content: center;
      align-items: center;
      flex-direction: column;
      position: absolute;
    }

    .full{
      display: flex;
      justify-content: center;
      align-items: center;
    }
  `,
  html: `
    <div class="container">
      <div class="row" >
        <div class="cel"> <div class="rolingtext full"></div>  </div>
        <div class="cel"><div class="svgborder full"></div> </div>
        <div class="cel"> <div class="rotatedline full"></div> </div>
        <div class="cel"> <div class="rotatedlinereveal full"></div> </div>
        <div class="cel"><div class="svgdraw full"></div> </div>
        <div class="cel"><div class="circle full"></div> </div>
        <div class="cel"><div class="logobox full"></div> </div>
        <div class="cel"><div class="textwriting full"></div> </div>
        <div class="cel"><div class="rightopacity full"></div> </div>
        <div class="cel"><div class="letterscale full"></div> </div>
        <div class="cel"><div class="circulartext full"></div> </div>
      </div>
    </div>
  `,
  fonts: [
    {
      type: `google-font`,
      src: `https://fonts.googleapis.com/css2?family=Rubik+Mono+One&display=swap`
    },
    {
      type: `google-font`,
      src: `https://fonts.googleapis.com/css2?family=Righteous&display=swap`
    },
    {
      type: "google-font",
      src:
        "https://fonts.googleapis.com/css2?family=Commissioner:wght@100;200;300;400;500;600;700;800;900&display=swap"
    }
  ],
  host: document.getElementById("clip"),
  containerParams: {
    width: "1280px",
    height: "720px"
  }
});

const svgdraw = new Titles.SvgDraw(
  {
    duration: 3600,
    width: 900,
    size: "L",
    lineColor: "#ff0000",
    strokeDashOffset: 10000,
    StrokeDashArray: 10000,
    svg: imgsvg,
    stopOnLast: false
  },
  {
    selector: ".svgdraw"
  }
);

const rolinText = new Titles.RolingText(
  {
    duration: 6000,
    width: 450,
    size: "XL",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle1: "The",
    subTitle2: "MotorCortex Team",
    subTitle3: "Presents",
    stopOnLast: false,
    delayEnd: 1000,
    fontFamily: "Commissioner"
  },
  {
    selector: ".rolingtext"
  }
);

const rotatedlinereveal = new Titles.RotatadLineReveal(
  {
    duration: 5000,
    width: 800,
    size: "M",
    lineColor: "#ff0000",
    textColor: "#000",
    title: "Animetitles",
    subtitle: "MC Plugin",
    leftEnd: 100,
    stopOnLast: false,
    fontFamily: "Commissioner"
  },
  {
    selector: ".rotatedlinereveal"
  }
);

const rotatedline = new Titles.RotatedLine(
  {
    duration: 6000,
    width: 400,
    size: "XL",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle1: "Minimal",
    subTitle2: "Elegant",
    subTitle3: "Modern",
    title: "Title",
    stopOnLast: false,
    fontFamily: "Commissioner"
  },
  {
    selector: ".rotatedline"
  }
);

const svgborder = new Titles.SvgBorder(
  {
    duration: 5000,
    size: "L",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle: "Developed by",
    slogan: "-KMB-",
    title: "MotorCortex Team",
    stopOnLast: false,
    fontFamily: "Commissioner"
  },
  {
    selector: ".svgborder"
  }
);

const circle = new Titles.Circle(
  {
    fontSize: 40,
    circleColor: "#ff0000",
    textColor: "#000",
    title: "ANIMATED",
    subTitle: "TITLES",
    stopOnLast: false,
    fontFamily: "Commissioner",
    width: 500
  },
  {
    selector: ".circle"
  }
);

const logobox = new Titles.LogoBox(
  {
    size: 2,
    textColor: "#fff",
    bgColor: "#000",
    subTitle1: "TYPOGRAPHY",
    subTitle2: "ANIMATION",
    subTitle3: "BY",
    width: 200,
    stopOnLast: false,
    fontFamily: "Commissioner",
    logoUrl:
      "https://kissmybutton.github.io/motorcortex-animetitles/demo/assets/logo.png"
  },
  {
    selector: ".logobox"
  }
);

const RightOpacity = new Titles.RightOpacity(
  {
    width: 1000,
    height: 250,
    text: "RightOpacity",
    color: "#0000",
    fontSize: 100,
    stroke: true,
    strokeSize: 2,
    strokeColor: "rebeccapurple",
    fontFamily: "Rubik Mono One",
    exitTime: 1000
  },
  {
    selector: ".rightopacity"
  }
);

const rightopacityOpacity = new Anime.Anime(
  {
    animatedAttrs: {
      opacity: 1
    },
    initialValues: {
      opacity: 0
    }
  },
  {
    duration: 1,
    selector: ".rightopacity"
  }
);

const LetterScale = new Titles.LetterScale(
  {
    width: 900,
    height: 900,
    color: "rebeccapurple",
    text: "LetterScale",
    fontSize: 100,
    stroke: false,
    strokeSize: 2,
    strokeColor: "rebeccapurple",
    fontFamily: "Righteous",
  },
  {
    selector: ".letterscale"
  }
);

const LetterScaleOpacity = new Anime.Anime(
  {
    animatedAttrs: {
      opacity: 1
    },
    initialValues: {
      opacity: 0
    }
  },
  {
    duration: 1,
    selector: ".letterscale"
  }
);
const LetterScaleOpacityBack = new Anime.Anime(
  {
    animatedAttrs: {
      opacity: 0
    }
  },
  {
    duration: 1,
    selector: ".letterscale"
  }
);

const CircularText = new Titles.CircularText(
  {
    width: 500,
    height: 500,
    color: "#ff0000",
    text: "CircularText - AnimeTitles - MotorCortex - Plugin -",
    fontSize: 10,
    viewBox: 100,
    path: 38,
    fill: "#ff000000",    
    fontFamily: "Righteous"
  },
  {
    selector: ".circulartext",
    repeats: 2,
  }
);

const CircularTextOpacity = new Anime.Anime(
  {
    animatedAttrs: {
      opacity: 1
    },
    initialValues: {
      opacity: 0
    }
  },
  {
    duration: 1,
    selector: ".circulartext"
  }
);

clip.addIncident(rolinText, 0);
clip.addIncident(rotatedlinereveal, 7000);
clip.addIncident(svgborder, 11025);
clip.addIncident(rotatedline, 16025);
clip.addIncident(circle, 22025);
clip.addIncident(logobox, 26225);
clip.addIncident(svgdraw, 29725);
clip.addIncident(rightopacityOpacity, 35845);
clip.addIncident(RightOpacity, 35846);
clip.addIncident(LetterScaleOpacity, clip.calculatedDuration);
clip.addIncident(LetterScale, clip.calculatedDuration);
clip.addIncident(LetterScaleOpacityBack, clip.calculatedDuration);
clip.addIncident(CircularTextOpacity, clip.calculatedDuration);
clip.addIncident(CircularText, clip.calculatedDuration);

new Player({ clip });
